lib/tests/streamlit/elements/vega_charts_test.py

```diff
@@ -121,6 +159,59 @@ def test_empty_altair_chart_throws_error(self):
         with self.assertRaises(TypeError):
             st.altair_chart(use_container_width=True)
 
+    def test_dataset_names_stay_stable(self):
+        """Test that dataset names stay stable across multiple calls
+        with new Pandas objects containing the same data.
+        """
+        # Execution 1:
+        df = pd.DataFrame([["A", "B", "C", "D"], [28, 55, 43, 91]], index=["a", "b"]).T
+        chart = alt.Chart(df).mark_bar().encode(x="a", y="b")
+        st.altair_chart(chart)
+        chart_el_1 = self.get_delta_from_queue().new_element
+
+        # Execution 2 (recreate the same chart with new objects)
+        df = pd.DataFrame([["A", "B", "C", "D"], [28, 55, 43, 91]], index=["a", "b"]).T
+        chart = alt.Chart(df).mark_bar().encode(x="a", y="b")
+        st.altair_chart(chart)
+
+        chart_el_2 = self.get_delta_from_queue().new_element
+
+        # Make sure that there is one named dataset:
+        self.assertEqual(len(chart_el_1.arrow_vega_lite_chart.datasets), 1)
+        # The names should not have changes
+        self.assertEqual(
+            [dataset.name for dataset in chart_el_1.arrow_vega_lite_chart.datasets],
+            [dataset.name for dataset in chart_el_2.arrow_vega_lite_chart.datasets],
+        )
+        # The specs should also be the same:
```

Comment:
Does this mean that the two `st.altair_chart` widgets have the same computed id or what else is contributing to the altair_chart id?

Reply:
There isn't an element/widget ID yet, this will be added later with selections. This test mainly validates that the specs are the exact same if the same chart + dataframe data are executed multiple times. In the current state, that's not the case.